examples/plot_adversarial_mitigation.py

```diff
@@ -0,0 +1,366 @@
+# Copyright (c) Fairlearn contributors.
+# Licensed under the MIT License.
+
+"""
+================================================
+Mitigating Fairness using Adversarial Mitigation
+================================================
+"""
+# %%
+# This notebook demonstrates our implementation of the technique *Mitigating*
+# *Unwanted Biases with Adversarial Learning* as proposed by
+# `Zhang et al. 2018 <https://dl.acm.org/doi/pdf/10.1145/3278721.3278779>`_.
+#
+# In short, the authors take classic supervised learning setting in which
+# a predictor neural network is trained, and extend it with an adversarial
+# network that aims to predict the sensitive feature. Then, they train the
+# predictor not only to minimize its own loss, but also minimize the predictive
+# ability of the adversarial.
+#
+# In short, we provide an implementation that supports:
+#
+# - Any predictor neural network implemented in either PyTorch or Tensorflow2
+# - Classification or regression
+# - Multiple sensitive features
+# - Two fairness objectives: Demographic parity or Equalized Odds
+#
+# This implementation follows closely the API of an `Estimator` in :class:`sklearn`
+
+# %%
+# Example 1: Simple use case with UCI Adult Dataset
+# =================================================
+# Firstly, we cover the most basic application of adversarial mitigation.
+# We start by loading and preprocessing the dataset.
+#
+# For this example we choose the feature 'sex' as the sensitive feature.
+
+
+# Get dataset.
+from math import sqrt
+from fairlearn.metrics import MetricFrame, selection_rate, \
+    demographic_parity_difference
+from sklearn.metrics import accuracy_score
+from fairlearn.adversarial import AdversarialFairnessClassifier, \
+    AdversarialFairness
+from pandas import Series
+from numpy import number, random, mean
+from sklearn.compose import make_column_transformer, make_column_selector
+from sklearn.preprocessing import OneHotEncoder, StandardScaler
+from sklearn.datasets import fetch_openml
+from sklearn.model_selection import train_test_split
+import torch
+random.seed(123)
```

Comment:
Sure, but I wonder why? Is it because this rng may be used in more places at the same time?

Reply:
It's just a better practice to use a local rng rather than the global one. Since setting the global seed can also change the generated values in other unrelated files if they also use the global rng.